DataLoader.lua

```diff
@@ -5,10 +5,20 @@ function DataLoader:index(indices, inputs, targets, ...)
    error"Not Implemented"
 end
 
+
+
 function DataLoader:sample(batchsize, inputs, targets, ...)
-   self._indices = self._indices or torch.LongTensor()
-   self._indices:resize(batchsize):random(1,self:size())
-   return self:index(self._indices, inputs, targets, ...)
+    self._indices = self._indices or torch.LongTensor()
+    if ifTrain == true then
```

Comment:
The ifTrain logic doesn't fit in. For training use sample. For testing use sub or index.